src/device.c

```diff
@@ -1411,93 +1411,73 @@ size_t uvc_num_devices(uvc_context_t *ctx) {
-  case 1: {  /* VideoControl interface */
-    int found_entity = 0;
-    struct uvc_input_terminal *input_terminal;
-    struct uvc_processing_unit *processing_unit;
-
-    if (transfer->actual_length < 5) {
-      UVC_DEBUG("Short read of VideoControl status update (%d bytes)",
-		transfer->actual_length);
-      UVC_EXIT_VOID();
-      return;
-    }
+  if (originator == 0) {
+    UVC_DEBUG("Unhandled update from VC interface");
+    UVC_EXIT_VOID();
+    return;  /* @todo VideoControl virtual entity interface updates */
+  }
 
-    event = transfer->buffer[2];
-    selector = transfer->buffer[3];
+  if (event != 0) {
+    UVC_DEBUG("Unhandled VC event %d", (int) event);
+    UVC_EXIT_VOID();
+    return;
+  }
 
-    if (originator == 0) {
-      UVC_DEBUG("Unhandled update from VC interface");
-      UVC_EXIT_VOID();
-      return;  /* @todo VideoControl virtual entity interface updates */
-    }
+  /* printf("bSelector: %d\n", selector); */
 
-    if (event != 0) {
-      UVC_DEBUG("Unhandled VC event %d", (int) event);
-      UVC_EXIT_VOID();
-      return;
+  DL_FOREACH(devh->info->ctrl_if.input_term_descs, input_terminal) {
+    if (input_terminal->bTerminalID == originator) {
+      status_class = UVC_STATUS_CLASS_CONTROL_CAMERA;
+      found_entity = 1;
+      break;
     }
+  }
 
-    /* printf("bSelector: %d\n", selector); */
-
-    DL_FOREACH(devh->info->ctrl_if.input_term_descs, input_terminal) {
-      if (input_terminal->bTerminalID == originator) {
-        status_class = UVC_STATUS_CLASS_CONTROL_CAMERA;
+  if (!found_entity) {
+    DL_FOREACH(devh->info->ctrl_if.processing_unit_descs, processing_unit) {
+      if (processing_unit->bUnitID == originator) {
+        status_class = UVC_STATUS_CLASS_CONTROL_PROCESSING;
         found_entity = 1;
         break;
       }
     }
-
-    if (!found_entity) {
-      DL_FOREACH(devh->info->ctrl_if.processing_unit_descs, processing_unit) {
-        if (processing_unit->bUnitID == originator) {
-          status_class = UVC_STATUS_CLASS_CONTROL_PROCESSING;
-          found_entity = 1;
-          break;
-        }
-      }
-    }
-
-    if (!found_entity) {
-      UVC_DEBUG("Got status update for unknown VideoControl entity %d",
-		(int) originator);
-      UVC_EXIT_VOID();
-      return;
-    }
-
-    attribute = transfer->buffer[4];
-    data = transfer->buffer + 5;
-    data_len = transfer->actual_length - 5;
-    break;
   }
-  case 2:  /* VideoStreaming interface */
-    UVC_DEBUG("Unhandled update from VideoStreaming interface");
+
+  if (!found_entity) {
+    UVC_DEBUG("Got status update for unknown VideoControl entity %d",
+  (int) originator);
     UVC_EXIT_VOID();
-    return;  /* @todo VideoStreaming updates */
+    return;
   }
 
-  UVC_DEBUG("Event: class=%d, event=%d, selector=%d, attribute=%d, data_len=%zd",
-    status_class, event, selector, attribute, data_len);
+  attribute = data[4];
+  content = data + 5;
+  content_len = len - 5;
```

Comment:
5 bytes are skipped already, as it was before these changes. But originator data is indeed lost.